cli/npm/managed/mod.rs

```diff
@@ -246,15 +254,17 @@ async fn snapshot_from_lockfile(
 /// An npm resolver where the resolution is managed by Deno rather than
 /// the user bringing their own node_modules (BYONM) on the file system.
 pub struct ManagedCliNpmResolver {
-  api: Arc<CliNpmRegistryApi>,
   fs: Arc<dyn FileSystem>,
   fs_resolver: Arc<dyn NpmPackageFsResolver>,
-  global_npm_cache: Arc<NpmCache>,
-  resolution: Arc<NpmResolution>,
+  http_client: Arc<HttpClient>,
```

Comment:
I'll work on ensuring this is in an ideal place later on. Just didnt' want to land a massive refactor in one go.